pkg/apis/planetscale/v2/vitesscluster_defaults.go

```diff
@@ -99,6 +99,31 @@ func DefaultVitessKeyspaceTemplate(keyspace *VitessKeyspaceTemplate) {
 	if keyspace.TurndownPolicy == "" {
 		keyspace.TurndownPolicy = VitessKeyspaceTurndownPolicyRequireIdle
 	}
+
+	for i := range keyspace.Partitionings {
+		partition := &keyspace.Partitionings[i]
+		defaultCustomPartitioning(partition.Custom)
+		defaultEqualPartitioning(partition.Equal)
+	}
+}
+
+func defaultCustomPartitioning(customPartition *VitessKeyspaceCustomPartitioning) {
+	if customPartition == nil {
+		return
+	}
+
+	for i := range customPartition.Shards {
+		shard := &customPartition.Shards[i]
```

Comment:
Super small nit, but I'm not sure this line buys us much in code clarity. Up to you but you can probably just one line this:
```suggestion
		DefaultVitessShardTemplate(&customPartition.Shards[i].VitessShardTemplate)
```